keadm/cmd/keadm/app/cmd/beta/config.go

```diff
@@ -0,0 +1,232 @@
+		if err := dockerPullImage(image); err != nil {
+			fmt.Printf("Failed to pull image %q: %v", image, err)
+			return fmt.Errorf("failed to pull image %q: %v", image, err)
+		}
+		fmt.Printf("Successfully pulled %s\n", image)
+	}
+	return nil
+}
+
+// AddImagesCommonConfigFlags adds the flags that configure keadm
+func AddImagesCommonConfigFlags(cmd *cobra.Command, cfg *Configuration) {
+	cmd.Flags().StringVar(&cfg.KubeEdgeVersion, cmdcommon.KubeEdgeVersion, cfg.KubeEdgeVersion,
+		`Use this key to decide which a specific KubeEdge version to be used.`,
+	)
+	cmd.Flags().StringVar(&cfg.ImageRepository, cmdcommon.ImageRepository, cfg.ImageRepository,
+		`Use this key to decide which image repository to pull images from.`,
+	)
+	cmd.Flags().StringVar(&cfg.Part, "part", cfg.Part,
+		"Use this key to set which part keadm will install: cloud part or edge part. If not set, keadm will list/pull all images used by both cloud part and edge part.")
+}
+
+// GetKubeEdgeImages returns a list of container images that related part expects to use
+func GetKubeEdgeImages(cfg *Configuration) []string {
+	var images []string
+
+	part := strings.ToLower(cfg.Part)
+
+	if part == "cloud" {
+		images = append(images, GetKubeEdgeCloudPartImages(cfg)...)
+	} else if part == "edge" {
+		images = append(images, GetKubeEdgeEdgePartImages(cfg)...)
+	} else {
+		// if not specified, will return all images used by both cloud part and edge part
+		images = append(images, GetKubeEdgeCloudPartImages(cfg)...)
+		images = append(images, GetKubeEdgeEdgePartImages(cfg)...)
+	}
+
+	return images
+}
+
+// GetKubeEdgeCloudPartImages returns a list of container images that Cloud part expects to use
+func GetKubeEdgeCloudPartImages(cfg *Configuration) []string {
+	images := []string{}
+
+	// Cloud part images
+	images = append(images, GetImage("admission", cfg))
+	images = append(images, GetImage("cloudcore", cfg))
+	images = append(images, GetImage("iptables-manager", cfg))
+	images = append(images, GetImage("installation-package", cfg))
+
+	return images
+}
+
+// GetKubeEdgeEdgePartImages returns a list of container images that Edge expects to use
+func GetKubeEdgeEdgePartImages(cfg *Configuration) []string {
+	images := []string{}
+
+	// Edge part images
+	images = append(images, GetPauseImage())
+	images = append(images, GetImage("installation-package", cfg))
+	images = append(images, "eclipse-mosquitto:1.6.15")
+
+	return images
+}
+
+// GetImage generates the image required for the KubeEdge
+func GetImage(image string, cfg *Configuration) string {
+	repoPrefix := cfg.GetImageRepository()
+	imageTag := cfg.KubeEdgeVersion
+
+	return fmt.Sprintf("%s/%s:%s", repoPrefix, image, imageTag)
+}
+
+func GetPauseImage() string {
+	return constants.DefaultPodSandboxImage
+}
+
+// dockerPullImage uses Docker to pull the image
+func dockerPullImage(image string) error {
+	var err error
+	for i := 0; i < PullImageRetry; i++ {
+		pullCommand := fmt.Sprintf("docker pull %s", image)
```

Comment:
Can open the new pr, we should use the sdk for `docker` or other runtime instead of docker cmd, can ref the pull image code in beta join.
/approve